src/logic/safe/transactions/gas.ts

```diff
@@ -61,59 +60,50 @@ export type TransactionExecutionEstimationProps = {
 }
 
 export const estimateGasForTransactionExecution = async ({
-  safeAddress,
-  safeVersion,
-  txRecipient,
-  txConfirmations,
-  txAmount,
-  txData,
-  operation,
-  from,
+  baseGas,
+  data,
   gasPrice,
   gasToken,
+  operation,
   refundReceiver,
+  safeInstance,
   safeTxGas,
-  approvalAndExecution,
-}: TransactionExecutionEstimationProps): Promise<number> => {
-  const safeInstance = getGnosisSafeInstanceAt(safeAddress, safeVersion)
-  // If it's approvalAndExecution we have to add a preapproved signature else we have all signatures
-  const sigs = generateSignaturesFromTxConfirmations(txConfirmations, approvalAndExecution ? from : undefined)
-
+  sigs,
+  to,
+  valueInWei,
+  safeAddress,
+  sender,
+}: TxArgs & { safeAddress: string }): Promise<number> => {
   const estimationData = safeInstance.methods
-    .execTransaction(txRecipient, txAmount, txData, operation, safeTxGas, 0, gasPrice, gasToken, refundReceiver, sigs)
+    .execTransaction(to, valueInWei, data, operation, safeTxGas, baseGas, gasPrice, gasToken, refundReceiver, sigs)
     .encodeABI()
 
   return calculateGasOf({
     data: estimationData,
-    from,
+    from: sender,
```

Comment:
why change to sender instead of using from?

Reply:
We realized that the type `TransactionExecutionEstimationProps` was basically the same as `TxArgs` but the fields have slightly different names. We wanted to unify this to make all the different Transaction types less confusing.